src/riverroute/mosart_histfile.F90

```diff
@@ -1195,115 +1217,124 @@ subroutine mosart_hist_Restart (ncid, flag, rdate)
+               call ncd_defvar(ncid=ncid_hist(t,f), varname='fexcl', xtype=ncd_char, &
+                    comment="Namelist item", &
+                    long_name="Fieldnames to exclude",  &
+                    dim1name='fname_lenp2', dim2name='max_flds' )
+
+               call ncd_defvar(ncid=ncid_hist(t,f), varname='nflds', xtype=ncd_int, &
+                    long_name="Number of fields on file", units="unitless",        &
+                    dim1name='scalar')
+               call ncd_defvar(ncid=ncid_hist(t,f), varname='ntimes', xtype=ncd_int, &
+                    long_name="Number of time steps on file", units="time-step",     &
+                    dim1name='scalar')
+               call ncd_defvar(ncid=ncid_hist(t,f), varname='is_endhist', xtype=ncd_log, &
+                    long_name="End of history file", dim1name='scalar')
+               call ncd_defvar(ncid=ncid_hist(t,f), varname='begtime', xtype=ncd_double, &
+                    long_name="Beginning time", units="time units",     &
+                    dim1name='scalar')
+
+               call ncd_defvar(ncid=ncid_hist(t,f), varname='hpindex', xtype=ncd_int, &
+                    long_name="History pointer index", units="unitless",     &
+                    dim1name='max_nflds' )
+
+               call ncd_defvar(ncid=ncid_hist(t,f), varname='avgflag', xtype=ncd_char, &
+                    long_name="Averaging flag", &
+                    units="A=Average, X=Maximum, M=Minimum, I=Instantaneous", &
+                    dim1name='len1', dim2name='max_nflds' )
+               call ncd_defvar(ncid=ncid_hist(t,f), varname='name', xtype=ncd_char, &
+                    long_name="Fieldnames",  &
+                    dim1name='fname_len', dim2name='max_nflds' )
+               call ncd_defvar(ncid=ncid_hist(t,f), varname='long_name', xtype=ncd_char, &
+                    long_name="Long descriptive names for fields", &
+                    dim1name='max_chars', dim2name='max_nflds' )
+               call ncd_defvar(ncid=ncid_hist(t,f), varname='units', xtype=ncd_char, &
+                    long_name="Units for each history field output", &
+                    dim1name='max_chars', dim2name='max_nflds' )
+               call ncd_enddef(ncid_hist(t,f))
+
+            end do file_loop1
+         end do tape_loop1
 
          RETURN
 
          !================================================
       else if (flag == 'write') then
          !================================================
          ! Add history filenames to master restart file
-         do t = 1,ntapes
-            call ncd_io('locfnh',  locfnh(t),  'write', ncid, nt=t)
-            call ncd_io('locfnhr', locfnhr(t), 'write', ncid, nt=t)
-         end do
+         tape_loop2: do t = 1, ntapes
+            file_loop2: do f = 1, maxsplitfiles
+               call ncd_io('locfnh',  my_locfnh,  'write', ncid, nt=counter)
+               call ncd_io('locfnhr', my_locfnhr, 'write', ncid, nt=counter)
```

Comment:
Note to self @slevis-lmwg 
I have not declared my_locfnh* and countr, yet. See how I did this in the clm equivalent.

Reply:
Troubleshooting with this test
`SMS_D_Ld5.f10_f10_mg37.I1850Clm60Sp.izumi_nag.mosart-default.0606-164622iz_nag`

Reply:
Resolving this conversation and returning to the full mosart test-suite.